core/variant/array.cpp

```diff
@@ -556,6 +556,26 @@ Array::Array() {
 	_p->refcount.init();
 }
 
+Array::Array(int size) {
+	_p = memnew(ArrayPrivate);
+	_p->refcount.init();
+	resize(size);
+}
+
+Array::Array(int size, const Variant &fill) {
+	_p = memnew(ArrayPrivate);
+	_p->refcount.init();
+	resize(size);
+	if (fill.get_type() < Variant::DICTIONARY) {
+		for (int i = 0; i < size; i++)
+			set(i, fill);
+	} else {
+		for (int i = 0; i < size - 1; i++)
+			set(i, fill.duplicate(true));
+		set(size - 1, fill);
```

Comment:
I think duplicating `fill` for all of the entries besides the last one might result in an unexpected behaviour. I'd say be consistent and duplicate `fill` for all entries.
Also it should be documented when `fill` is being duplicated.